### PR TITLE
Deprecate Morphia#useBulkWriteOperations

### LINKED ISSUE #1052: Deprecate Morphia#useBulkWriteOperations
This field provides no real value since only bulk inserts are exposed and not mixed operations.  In such scenarios bulk write or not makes no difference.  If a future API exposes the ability to do mixed operations in bulk, this field would be redundant anyway.

### PR DESCRIPTION
**Pull request by evanchooly**
fixes #1052

### CODE REVIEW

File: morphia/src/main/java/org/mongodb/morphia/Morphia.java [hunk 1]
```diff
@@ -160,19 +165,23 @@ public boolean isMapped(final Class entityClass) {
...
      * Configures Morphia to use bulk writes.  Only useful with MongoDB 2.6+.
      *
      * @param useBulkWriteOperations true if Morphia should use bulk writes
+     * @see <a href="https://github.com/mongodb/morphia/issues/1052">Issue #1052</a>
+     * @deprecated Setting this value has no value functionally or performance-wise.
      */
+    @Deprecated
     public void setUseBulkWriteOperations(final boolean useBulkWriteOperations) {
-        this.useBulkWriteOperations = useBulkWriteOperations;
+        LOG.warning("Setting useBulkWriteOperations has no value functionally or performance-wise.");
```
Comment (jyemin) [inline morphia/src/main/java/org/mongodb/morphia/Morphia.java]: I think the deprecate is sufficient.  Why also log a warning?

File: morphia/src/main/java/org/mongodb/morphia/Morphia.java [hunk 2]
```diff
@@ -160,19 +165,23 @@ public boolean isMapped(final Class entityClass) {
     }
 
     /**
-     * @return true if Morphia should use bulk write operations.  Only useful with MongoDB 2.6+.
+     * @return false.  Setting this value has no value functionally or performance-wise.
+     * @see <a href="https://github.com/mongodb/morphia/issues/1052">Issue #1052</a>
      */
     public boolean isUseBulkWriteOperations() {
```
Comment (jyemin) [inline morphia/src/main/java/org/mongodb/morphia/Morphia.java]: This should also be deprecated then

File: morphia/src/main/java/org/mongodb/morphia/Morphia.java (hunk 1 above)
Comment (evanchooly) [inline morphia/src/main/java/org/mongodb/morphia/Morphia.java]: Visibility mainly.  An ongoing warning that things have changed.  Figured it was the polite/correct thing to do.

File: morphia/src/main/java/org/mongodb/morphia/Morphia.java (hunk 2 above)
Comment (evanchooly) [inline morphia/src/main/java/org/mongodb/morphia/Morphia.java]: Meant to and missed it.

File: morphia/src/main/java/org/mongodb/morphia/Morphia.java (hunk 1 above)
Comment (jyemin) [inline morphia/src/main/java/org/mongodb/morphia/Morphia.java]: Please pull it out then, as it's inconsistent with all the other deprecated methods, none of which log a warning.

Comment (evanchooly) [inline morphia/src/main/java/org/mongodb/morphia/Morphia.java]: done

**Comment by jyemin** [comment]
lgtm

**Pull request merged**